pysmt/factory.py

```diff
@@ -97,12 +100,60 @@ def get_solver(self, quantified=False, name=None, logic=None):
             closer_logic = get_closer_logic(SolverClass.LOGICS, logic)
             return SolverClass(environment=self.environment,
                                logic=closer_logic,
-                               options=None)
+                               user_options={"generate_models":True})
         else:
             raise NoSolverAvailableError("No solver is available for logic %s" %\
                                          logic)
 
 
+    def get_unsat_core_solver(self, quantified=False, name=None,
```

Comment:
Maybe we can factor this function together with the one for picking a regular solver. Ideally, we are looking for a solver with some options, inside a set of solvers. In one case the set of solvers is all_solvers, in the other it is all_unsat_core_solvers.